src/extensions/extension-discovery.ts

```diff
@@ -125,15 +125,15 @@ export class ExtensionDiscovery extends Singleton {
   /**
    * Initializes the class and setups the file watcher for added/removed local extensions.
    */
-  async init() {
+  async init(): Promise<void> {
```

Comment:
Explicit return types 👍

Reply:
Would you be in favour of eventually requiring these using eslint?